pkg/asset/manifests/proxy.go

```diff
@@ -0,0 +1,139 @@
+package manifests
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+var proxyCfgFilename = filepath.Join(manifestDir, "cluster-proxy-01-config.yml")
+
+// Proxy generates the cluster-proxy-*.yml files.
+type Proxy struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*Proxy)(nil)
+
+// Name returns a human-friendly name for the asset.
+func (*Proxy) Name() string {
+	return "Proxy Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*Proxy) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&Networking{},
+	}
+}
+
+// Generate generates the Proxy config and its CRD.
+func (p *Proxy) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	network := &Networking{}
+	dependencies.Get(installConfig, network)
+
+	config := &configv1.Proxy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: configv1.SchemeGroupVersion.String(),
+			Kind:       "Proxy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+			// not namespaced
+		},
+	}
+
+	if installConfig.Config.Proxy != nil {
+		config.Spec = configv1.ProxySpec{
+			HTTPProxy:  installConfig.Config.Proxy.HTTPProxy,
+			HTTPSProxy: installConfig.Config.Proxy.HTTPSProxy,
+			NoProxy:    installConfig.Config.Proxy.NoProxy,
+		}
+	}
+
+	if config.Spec.HTTPProxy != "" || config.Spec.HTTPSProxy != "" {
+		config.Status = configv1.ProxyStatus{
+			HTTPProxy:  installConfig.Config.Proxy.HTTPProxy,
+			HTTPSProxy: installConfig.Config.Proxy.HTTPSProxy,
+			NoProxy:    CreateNoProxy(installConfig),
+		}
+	}
+
+	configData, err := yaml.Marshal(config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s manifests from InstallConfig", p.Name())
+	}
+
+	p.FileList = []*asset.File{
+		{
+			Filename: proxyCfgFilename,
+			Data:     configData,
+		},
+	}
+
+	return nil
+}
+
+// CreateNoProxy adds platform-specific NO_PROXY values to the user-provided values
+// Platform values are: serviceCIDR, podCIDR, localhost, 127.0.0.1, api.clusterdomain,
+// api-int.clusterdomain, etcd-idx.clusterdomain
+func CreateNoProxy(installConfig *installconfig.InstallConfig) string {
+	defaults := []string{
+		"127.0.0.1",
+		"localhost",
+	}
+
+	defaults = append(defaults, installConfig.Config.Networking.ServiceNetwork[0].String())
```

Comment:
this should be from the configv1.Network from Networking asset compared to currently the InstallConfig